lua/neogit/popups/tag/actions.lua

```diff
@@ -0,0 +1,161 @@
+local M = {}
+local git = require("neogit.lib.git")
+local client = require("neogit.client")
+local utils = require("neogit.lib.util")
+local FuzzyFinderBuffer = require("neogit.buffers.fuzzy_finder")
+
+--- Gets revisions
+---@return table
+local function get_revisions()
+  local revisions = git.cli["for-each-ref"].format('"%(refname:short)"').call():trim().stdout
+  for i, str in ipairs(revisions) do
+    revisions[i] = string.sub(str, 2, -2)
+  end
+  return revisions
+end
+
+--- Outputs a list of tags locally
+---@return table|nil List of tags.
+local function list_tags()
+  local revisions = git.cli.tag.args("-l").call():trim().stdout
+  if #revisions == 0 then
+    return nil
+  end
+  return revisions
+end
+
+function M.create_tag(popup)
+  local value = vim.fn.input { prompt = "Tag name: " }
+  if not value then
+    return
+  end
+  local options = get_revisions()
+  local selected_branch = FuzzyFinderBuffer.new(options):open_async()
+  if not selected_branch then
+    return
+  end
+  local args = popup:get_arguments()
+  if vim.tbl_count(args) > 0 and not vim.tbl_contains(args, "--annotate") then
+    table.insert(args, "--annotate")
+  end
+  client.wrap(git.cli.tag.arg_list({ value, selected_branch }).apply_flags(args), {
+    autocmd = "NeogitTagComplete",
+    msg = {
+      success = "Added tag " .. value .. " on " .. selected_branch,
+      fail = "Failed to add tag " .. value .. " on " .. selected_branch,
+    },
+  })
+end
+
+--- Create a release tag for `HEAD'.
+---@param _ table
+function M.create_release(_) end
+
+--- Delete one or more tags.
+--- If there are multiple tags then offer to delete those.
+--- Otherwise prompt for a single tag to be deleted.
+--- git tag -d TAGS
+---@param _ table
+function M.delete(_)
+  local a = require("plenary.async")
+  local notif = require("neogit.lib.notification")
+  local options = list_tags()
+  local tags = {}
+
+  if options ~= nil then
+    local selected_tags = FuzzyFinderBuffer.new(options):open_async { allow_multi = true }
+    if #selected_tags == 0 then
+      return
+    end
+    tags = selected_tags
+  else
+    local tag_input = vim.fn.input { prompt = "Tag name:" }
+    if not tag_input or tag_input == "" then
+      return
+    end
+    table.insert(tags, tag_input)
+  end
+
+  local result = git.cli.tag.arg_list({ "-d", unpack(tags) }).call():trim()
```

Comment:
Add `--delete` flag to cli builder. Could also create `git.tag.delete(tags)` function that returns true/false on success/failure.